src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,174 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public static final String EMPTY_LIST_MESSAGE = "List is empty";
+    public static final String INVALID_INDEX_MESSAGE = "Invalid index: ";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> node = new Node<>(tail, value, null);
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        validateAddIndex(index);
+
+        if (index == size) {
+            add(value);
+        } else {
```

Comment:
I suggest add 3 if`s in this method: 1 if head, 2 if tail, 3 if in middle in will make code more readable

Reply:
I would also suggest using `return` instead of `else`, and I do not see the changes that Andrii proposed, so please see what we can do about this or leave a comment on why you think otherwise

Reply:
I was trying to implement Andrii suggestionfor more then an hour, but tets were failing every tyme...
Maybe i dont get it right...

Reply:
> I would also suggest using `return` instead of `else`, and I do not see the changes that Andrii proposed, so please see what we can do about this or leave a comment on why you think otherwise

As i understend u, i should do like this: (this solution is passing all tests)
```@Override
    public void add(T value, int index) {
        validateIndexOnAdd(index);

        if (index == size) {
            add(value);
            return;
        }
        
        Node<T> nodeAtIndex = findNodeByIndex(index);
        Node<T> node = new Node<>(nodeAtIndex.prev, value, nodeAtIndex);

        if (index == 0) {
            head = node;
        } else {
            nodeAtIndex.prev.next = node;
        }
        nodeAtIndex.prev = node;
        size++;
    }